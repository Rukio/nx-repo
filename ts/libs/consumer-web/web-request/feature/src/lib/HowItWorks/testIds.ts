export const HOW_IT_WORKS_TEST_IDS = {
  TITLE: 'how-it-works-title',
  SUBTITLE: 'how-it-works-subtitle',
  BOOK_APPOINTMENT_BTN: 'book-appointment-btn',
};
