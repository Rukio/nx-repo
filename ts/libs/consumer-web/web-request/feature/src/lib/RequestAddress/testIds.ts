export const REQUEST_ADDRESS_TEST_IDS = {
  RETURN_HOME_BUTTON: 'return-home-button',
  LOCATION_CONTINUE_BUTTON: 'location-continue-button',
  LOCATION_HEADER: 'location-header',
  LOCATION_CONFIRM_MESSAGE: 'location-confirm-message',
  GOOGLE_MAP_CONTAINER: 'google-map-container',
  IN_SERVICE_MESSAGE: 'in-service-message',
  OUT_OF_AREA_MESSAGE: 'out-of-area-message',
  MARKET_CLOSED_MESSAGE: 'market-closed-message',
  LIMITED_APPOINTMENT_WARNING: 'limited-appointment-warning',
};
