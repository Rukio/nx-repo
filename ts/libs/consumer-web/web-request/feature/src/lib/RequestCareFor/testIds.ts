export const REQUEST_CARE_FOR_TEST_IDS = {
  REQUESTING_CARE_HEADER: 'requesting-care-header',
  CONTINUE_BUTTON: 'continue-button',
  getCareForButtonTestId: (option: string) =>
    `${option.toLowerCase().replace(/\s/g, '-')}-option-button`,
};
