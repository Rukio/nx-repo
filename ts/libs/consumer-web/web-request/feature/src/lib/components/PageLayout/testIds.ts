export const PAGE_LAYOUT_TEST_IDS = {
  MISSED_REQUIRED_FIELDS_TITLE: 'missed-required-fields-title',
  MISSED_REQUIRED_FIELDS_LIST: 'missed-required-fields-list',
  MISSED_REQUIRED_FIELDS_LIST_ITEM: 'missed-required-fields-list-item',
  DISPATCH_HEALTH_LOGO: 'dispatch-health-logo',
  REQUEST_PROGRESS_BAR: 'request-progress-bar',
  getBackLinkTestId: (text: string) =>
    `${text.toLowerCase().replace(/\s/g, '-')}-back-link`,
  EMERGENCY_INFO_HEADER: 'emergency-info-header',
  SUCCESS_MESSAGE_HEADER: 'success-message-header',
  SUBMIT_VISIT_BUTTON: 'submit-visit-button',
  EMERGENCY_INFO_FOOTER: 'emergency-info-footer',
  PP_TOC_HEADER: 'pp-toc-header',
  PRIVATE_POLICY_LINK: 'private-policy-link',
  TERMS_OF_SERVICE_LINK: 'terms-of-service-link',
  GRECAPTCHA_TEXT: 'grecaptcha-text',
};
