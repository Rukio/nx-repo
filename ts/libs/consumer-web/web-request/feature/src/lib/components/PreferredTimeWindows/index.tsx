export { default as PreferredTimeWindows } from './PreferredTimeWindows';
export * from './PreferredTimeWindows';
export * from './testIds';
