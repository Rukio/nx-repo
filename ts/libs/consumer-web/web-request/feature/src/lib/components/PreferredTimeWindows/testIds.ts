export const PREFERRED_TIME_WINDOWS_TEST_IDS = {
  ALERT: 'preferred-time-windows-alert',
  getDayOptionsGroupTestId: (day: string) =>
    `preferred-time-windows-${day}-options-group`,
  getDayOptionsHeaderTestId: (day: string) =>
    `preferred-time-windows-${day}-options-header`,
};
