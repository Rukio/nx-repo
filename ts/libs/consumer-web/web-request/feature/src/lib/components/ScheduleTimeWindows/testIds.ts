export const SCHEDULE_TIME_WINDOWS_TEST_IDS = {
  ALERT: 'schedule-time-windows-alert',
  AVAILABILITY_MESSAGE: 'schedule-time-windows-availability-message',
  DAY_TOGGLE_BUTTON_GROUP: 'schedule-time-windows-day-toggle-button-group',
  getDayToggleButtonTestId: (day: string) =>
    `schedule-time-windows-day-toggle-button-${day}`,
  MIN_TIME_RANGE_ERROR_MESSAGE:
    'schedule-time-windows-min-time-range-error-message',
};
