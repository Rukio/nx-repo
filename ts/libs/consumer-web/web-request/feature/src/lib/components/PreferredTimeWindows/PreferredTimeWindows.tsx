import { FC, MouseEvent } from 'react';
import {
  Typography,
  ToggleButton,
  ToggleButtonGroup,
  Alert,
  colorManipulator,
  makeSxStyles,
} from '@*company-data-covered*/design-system';
import { PatientPreferredEta } from '@*company-data-covered*/consumer-web-types';
import { SessionStorageKeys } from '../../constants';
import { PREFERRED_TIME_WINDOWS_TEST_IDS } from './testIds';
import {
  DaysEnum,
  getTomorrowDate,
  TimePeriod,
  getTimePeriod,
  isSameTime,
  isFutureDate,
  getMonthDayFormattedDate,
} from '../../utils';

export interface PreferredTimeWindowsProps {
  isFutureEta: boolean;
  requestPreferredEta: PatientPreferredEta | undefined;
  onChangeRequestPreferredEta: (option: {
    patientPreferredEtaStart: Date;
    patientPreferredEtaEnd: Date;
  }) => void;
}

const makeStyles = () =>
  makeSxStyles({
    toggleButtonGroup: {
      '& .MuiToggleButtonGroup-grouped': {
        my: 1.5,

        '&:not(:first-of-type), &:first-of-type': {
          borderRadius: 1,
          border: 1,
        },

        '&.Mui-selected, &.Mui-selected:hover': (theme) => ({
          backgroundColor: colorManipulator.alpha(
            theme.palette.primary.main,
            0.08
          ),
          borderColor: theme.palette.primary.main,
          color: theme.palette.primary.main,
        }),
      },
    },
  });

interface TimeOption {
  title: string;
  start: Date;
  end: Date;
}

interface DayOption {
  day: DaysEnum;
  time: TimeOption;
}

const getPreferredEta = (dayOption: DayOption) => {
  if (dayOption.day === DaysEnum.tomorrow) {
    return {
      patientPreferredEtaStart: getTomorrowDate(dayOption.time.start),
      patientPreferredEtaEnd: getTomorrowDate(dayOption.time.end),
    };
  }

  return {
    patientPreferredEtaStart: dayOption.time.start,
    patientPreferredEtaEnd: dayOption.time.end,
  };
};

const PreferredTimeWindows: FC<PreferredTimeWindowsProps> = ({
  isFutureEta,
  requestPreferredEta,
  onChangeRequestPreferredEta,
}) => {
  const timeOptions = {
    morning: {
      title: `${TimePeriod.Morning} (7am-12pm)`,
      ...getTimePeriod(TimePeriod.Morning),
    },
    afternoon: {
      title: `${TimePeriod.Afternoon} (12-5pm)`,
      ...getTimePeriod(TimePeriod.Afternoon),
    },
    evening: {
      title: `${TimePeriod.Evening} (5-10pm)`,
      ...getTimePeriod(TimePeriod.Evening),
    },
  };

  const todayOptions: DayOption[] = [
    { day: DaysEnum.today, time: timeOptions.morning },
    { day: DaysEnum.today, time: timeOptions.afternoon },
    { day: DaysEnum.today, time: timeOptions.evening },
  ];

  const tomorrowOptions: DayOption[] = [
    { day: DaysEnum.tomorrow, time: timeOptions.morning },
    { day: DaysEnum.tomorrow, time: timeOptions.afternoon },
    { day: DaysEnum.tomorrow, time: timeOptions.evening },
  ];

  const onChangeRequestFor = (_event: MouseEvent, value: DayOption) => {
    const option = getPreferredEta(value);
    const isSameOptionSelected =
      isSameTime(
        requestPreferredEta?.patientPreferredEtaStart,
        option.patientPreferredEtaStart
      ) &&
      isSameTime(
        requestPreferredEta?.patientPreferredEtaEnd,
        option.patientPreferredEtaEnd
      );

    if (isSameOptionSelected) {
      return;
    }

    onChangeRequestPreferredEta(option);
    window.sessionStorage.setItem(
      SessionStorageKeys.PreferredEtaStart,
      option.patientPreferredEtaStart.toString()
    );
  };

  const renderTimeOptions = (options: DayOption[], showTomorrow?: boolean) => {
    const styles = makeStyles();

    return (
      <ToggleButtonGroup
        exclusive
        orientation="vertical"
        fullWidth
        data-testid={PREFERRED_TIME_WINDOWS_TEST_IDS.getDayOptionsGroupTestId(
          options[0].day
        )}
        onChange={onChangeRequestFor}
        sx={styles.toggleButtonGroup}
      >
        {options.map((option) => {
          const startTimeOption = showTomorrow
            ? getTomorrowDate(option.time.start)
            : option.time.start;
          const endTimeOption = showTomorrow
            ? getTomorrowDate(option.time.end)
            : option.time.end;

          return (
            <ToggleButton
              key={`${option.day}-${option.time.title}`}
              selected={
                isFutureEta &&
                isSameTime(
                  requestPreferredEta?.patientPreferredEtaStart,
                  startTimeOption
                ) &&
                isSameTime(
                  requestPreferredEta?.patientPreferredEtaEnd,
                  endTimeOption
                )
              }
              value={option}
              disabled={!isFutureDate(endTimeOption)}
            >
              {option.time.title}
            </ToggleButton>
          );
        })}
      </ToggleButtonGroup>
    );
  };

  return (
    <>
      <Alert
        severity="warning"
        message="Appointments are booking fast! Select your preferred time window for a visit."
        data-testid={PREFERRED_TIME_WINDOWS_TEST_IDS.ALERT}
        sx={{ mt: 3 }}
      />
      <Typography
        variant="h5"
        sx={{ mt: 3 }}
        data-testid={PREFERRED_TIME_WINDOWS_TEST_IDS.getDayOptionsHeaderTestId(
          DaysEnum.today
        )}
      >
        Today ({getMonthDayFormattedDate(new Date())})
      </Typography>
      {renderTimeOptions(todayOptions)}
      <Typography
        variant="h5"
        sx={{ mt: 4 }}
        data-testid={PREFERRED_TIME_WINDOWS_TEST_IDS.getDayOptionsHeaderTestId(
          DaysEnum.tomorrow
        )}
      >
        Tomorrow ({getMonthDayFormattedDate(getTomorrowDate())})
      </Typography>
      {renderTimeOptions(tomorrowOptions, true)}
    </>
  );
};

export default PreferredTimeWindows;
