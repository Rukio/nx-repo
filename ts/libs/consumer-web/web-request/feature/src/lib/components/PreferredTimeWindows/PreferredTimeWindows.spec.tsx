import { render, screen } from '../../../testUtils';
import { within } from '@testing-library/react';
import dayjs from 'dayjs';
import PreferredTimeWindows, {
  PreferredTimeWindowsProps,
} from './PreferredTimeWindows';
import { PREFERRED_TIME_WINDOWS_TEST_IDS } from './testIds';
import { TimePeriod, DaysEnum } from '../../utils';

jest.mock('statsig-js', () => ({
  ...jest.requireActual('statsig-js'),
  logEvent: jest.fn(),
  checkGate: jest.fn(),
  getExperiment: () => ({
    get: (_name: string, fallback: unknown) => fallback,
  }),
  getConfig: () => ({
    get: (_name: string, fallback: unknown) => fallback,
  }),
}));

const TIME_OPTIONS = [
  {
    timePeriod: TimePeriod.Morning,
    title: `${TimePeriod.Morning} (7am-12pm)`,
  },
  {
    timePeriod: TimePeriod.Afternoon,
    title: `${TimePeriod.Afternoon} (12-5pm)`,
  },
  {
    timePeriod: TimePeriod.Evening,
    title: `${TimePeriod.Evening} (5-10pm)`,
  },
];

const defaultProps: PreferredTimeWindowsProps = {
  isFutureEta: true,
  requestPreferredEta: undefined,
  onChangeRequestPreferredEta: jest.fn(),
};

const setup = (props: Partial<PreferredTimeWindowsProps> = {}) => {
  return render(<PreferredTimeWindows {...defaultProps} {...props} />);
};

describe('<RequestPreferredTime />', () => {
  beforeEach(() => {
    jest.clearAllMocks();
    jest.useFakeTimers().setSystemTime(new Date('2020-04-30T07:00:00.000Z'));
  });

  afterEach(() => {
    jest.runOnlyPendingTimers();
    jest.useRealTimers();
  });

  it('RequestPreferredTime snapshot', () => {
    const { asFragment } = setup();
    expect(asFragment()).toMatchSnapshot();
  });

  it('should display correct dates in the titles', () => {
    setup();

    const preferredTimeAlert = screen.getByTestId(
      PREFERRED_TIME_WINDOWS_TEST_IDS.ALERT
    );
    expect(preferredTimeAlert).toHaveTextContent(
      'Appointments are booking fast! Select your preferred time window for a visit.'
    );

    const todayOptionsHeader = screen.getByTestId(
      PREFERRED_TIME_WINDOWS_TEST_IDS.getDayOptionsHeaderTestId(DaysEnum.today)
    );
    const tomorrowOptionsHeader = screen.getByTestId(
      PREFERRED_TIME_WINDOWS_TEST_IDS.getDayOptionsHeaderTestId(
        DaysEnum.tomorrow
      )
    );

    const today = `${dayjs().format('M')}/${dayjs().format('DD')}`;
    const tomorrow = `${dayjs().add(1, 'day').format('M')}/${dayjs()
      .add(1, 'day')
      .format('DD')}`;

    expect(todayOptionsHeader).toHaveTextContent(today);
    expect(tomorrowOptionsHeader).toHaveTextContent(tomorrow);
  });

  describe('validate time options', () => {
    beforeEach(() => {
      jest.clearAllMocks();
      // eslint-disable-next-line no-extend-native
      Date.prototype.getTimezoneOffset = jest.fn(() => 0);
    });

    it.each([
      [
        'all time options when 00:00:00',
        { mockLocalTime: '2020-04-30T00:00:00.000Z' },
      ],
    ])(
      `today's group time options - should enable %s local time`,
      (_, { mockLocalTime }) => {
        jest.useFakeTimers().setSystemTime(new Date(mockLocalTime));
        setup();

        const todayOptionsGroup = screen.getByTestId(
          PREFERRED_TIME_WINDOWS_TEST_IDS.getDayOptionsGroupTestId(
            DaysEnum.today
          )
        );

        TIME_OPTIONS.forEach((option) => {
          const timeOption = within(todayOptionsGroup).getByText(option.title);
          expect(timeOption).toBeEnabled();
        });
      }
    );

    it.each([
      [
        'Morning time option when 12:00:00',
        {
          mockLocalTime: '2020-04-30T12:00:00.000Z',
          disabledTimeOptionsCount: 1,
        },
      ],
      [
        'Morning time option when 16:59:00',
        {
          mockLocalTime: '2020-04-30T16:59:00.000Z',
          disabledTimeOptionsCount: 1,
        },
      ],
      [
        'Morning and Afternoon time options when 17:00:00',
        {
          mockLocalTime: '2020-04-30T17:00:00.000Z',
          disabledTimeOptionsCount: 2,
        },
      ],
    ])(
      `today's group time options - should disable %s local time`,
      (_, { mockLocalTime, disabledTimeOptionsCount }) => {
        jest.useFakeTimers().setSystemTime(new Date(mockLocalTime));
        setup();

        const todayOptionsGroup = screen.getByTestId(
          PREFERRED_TIME_WINDOWS_TEST_IDS.getDayOptionsGroupTestId(
            DaysEnum.today
          )
        );

        for (let i = 0; i < disabledTimeOptionsCount; i += 1) {
          const timeOption = within(todayOptionsGroup).getByText(
            TIME_OPTIONS[i].title
          );
          expect(timeOption).toBeDisabled();
        }
      }
    );

    it("tomorrow's group time options - should enable all time options", () => {
      setup();

      const tomorrowGroup = screen.getByTestId(
        PREFERRED_TIME_WINDOWS_TEST_IDS.getDayOptionsGroupTestId(
          DaysEnum.tomorrow
        )
      );

      TIME_OPTIONS.forEach((option) =>
        expect(within(tomorrowGroup).getByText(option.title)).toBeEnabled()
      );
    });
  });
});
