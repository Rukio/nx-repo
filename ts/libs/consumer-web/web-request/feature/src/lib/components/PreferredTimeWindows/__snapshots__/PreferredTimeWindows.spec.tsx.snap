// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RequestPreferredTime /> RequestPreferredTime snapshot 1`] = `
<DocumentFragment>
  .emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: none;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  background-color: rgb(255, 244, 229);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 6px 16px;
  color: rgb(102, 60, 0);
  margin-top: 24px;
}

.emotion-0 .MuiAlert-icon {
  color: #ff9800;
}

.emotion-1 {
  margin-right: 12px;
  padding: 7px 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 22px;
  opacity: 0.9;
}

.emotion-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: inherit;
}

.emotion-3 {
  padding: 8px 0;
  min-width: 0;
  overflow: auto;
}

<div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard emotion-0"
    data-testid="preferred-time-windows-alert"
    role="alert"
  >
    <div
      class="MuiAlert-icon emotion-1"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit emotion-2"
        data-testid="ReportProblemOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
        />
      </svg>
    </div>
    <div
      class="MuiAlert-message emotion-3"
    >
      Appointments are booking fast! Select your preferred time window for a visit.
    </div>
  </div>
  .emotion-0 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1.5rem;
  line-height: 1.334;
  letter-spacing: 0em;
  margin-top: 24px;
}

<h5
    class="MuiTypography-root MuiTypography-h5 emotion-0"
    data-testid="preferred-time-windows-today-options-header"
  >
    Today (4/30)
  </h5>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-radius: 4px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.emotion-0 .MuiToggleButtonGroup-grouped:not(:first-of-type) {
  margin-top: -1px;
  border-top: 1px solid transparent;
  border-top-left-radius: 0;
  border-top-right-radius: 0;
}

.emotion-0 .MuiToggleButtonGroup-grouped:not(:last-of-type) {
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
}

.emotion-0 .MuiToggleButtonGroup-grouped.Mui-selected+.MuiToggleButtonGroup-grouped.Mui-selected {
  border-top: 0;
  margin-top: 0;
}

.emotion-0 .MuiToggleButtonGroup-grouped {
  margin-top: 12px;
  margin-bottom: 12px;
}

.emotion-0 .MuiToggleButtonGroup-grouped:not(:first-of-type),
.emotion-0 .MuiToggleButtonGroup-grouped:first-of-type {
  border-radius: 4px;
  border: 1px solid;
}

.emotion-0 .MuiToggleButtonGroup-grouped.Mui-selected,
.emotion-0 .MuiToggleButtonGroup-grouped.Mui-selected:hover {
  background-color: rgba(25, 118, 210, 0.08);
  border-color: #1976d2;
  color: #1976d2;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  border-radius: 4px;
  padding: 11px;
  border: 1px solid rgba(0, 0, 0, 0.12);
  color: rgba(0, 0, 0, 0.54);
  width: 100%;
}

.emotion-1::-moz-focus-inner {
  border-style: none;
}

.emotion-1.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-1 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-1.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  border: 1px solid rgba(0, 0, 0, 0.12);
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-1:hover {
    background-color: transparent;
  }
}

.emotion-1.Mui-selected {
  color: rgba(0, 0, 0, 0.87);
  background-color: rgba(0, 0, 0, 0.08);
}

.emotion-1.Mui-selected:hover {
  background-color: rgba(0, 0, 0, 0.12);
}

@media (hover: none) {
  .emotion-1.Mui-selected:hover {
    background-color: rgba(0, 0, 0, 0.08);
  }
}

.emotion-2 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div
    class="MuiToggleButtonGroup-root MuiToggleButtonGroup-vertical MuiToggleButtonGroup-fullWidth emotion-0"
    data-testid="preferred-time-windows-today-options-group"
    role="group"
  >
    <button
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-fullWidth MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedVertical emotion-1"
      tabindex="0"
      type="button"
      value="[object Object]"
    >
      Morning (7am-12pm)
      <span
        class="MuiTouchRipple-root emotion-2"
      />
    </button>
    <button
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-fullWidth MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedVertical emotion-1"
      tabindex="0"
      type="button"
      value="[object Object]"
    >
      Afternoon (12-5pm)
      <span
        class="MuiTouchRipple-root emotion-2"
      />
    </button>
    <button
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-fullWidth MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedVertical emotion-1"
      tabindex="0"
      type="button"
      value="[object Object]"
    >
      Evening (5-10pm)
      <span
        class="MuiTouchRipple-root emotion-2"
      />
    </button>
  </div>
  .emotion-0 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1.5rem;
  line-height: 1.334;
  letter-spacing: 0em;
  margin-top: 32px;
}

<h5
    class="MuiTypography-root MuiTypography-h5 emotion-0"
    data-testid="preferred-time-windows-tomorrow-options-header"
  >
    Tomorrow (5/01)
  </h5>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  border-radius: 4px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.emotion-0 .MuiToggleButtonGroup-grouped:not(:first-of-type) {
  margin-top: -1px;
  border-top: 1px solid transparent;
  border-top-left-radius: 0;
  border-top-right-radius: 0;
}

.emotion-0 .MuiToggleButtonGroup-grouped:not(:last-of-type) {
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
}

.emotion-0 .MuiToggleButtonGroup-grouped.Mui-selected+.MuiToggleButtonGroup-grouped.Mui-selected {
  border-top: 0;
  margin-top: 0;
}

.emotion-0 .MuiToggleButtonGroup-grouped {
  margin-top: 12px;
  margin-bottom: 12px;
}

.emotion-0 .MuiToggleButtonGroup-grouped:not(:first-of-type),
.emotion-0 .MuiToggleButtonGroup-grouped:first-of-type {
  border-radius: 4px;
  border: 1px solid;
}

.emotion-0 .MuiToggleButtonGroup-grouped.Mui-selected,
.emotion-0 .MuiToggleButtonGroup-grouped.Mui-selected:hover {
  background-color: rgba(25, 118, 210, 0.08);
  border-color: #1976d2;
  color: #1976d2;
}

.emotion-1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  letter-spacing: 0.02857em;
  text-transform: uppercase;
  border-radius: 4px;
  padding: 11px;
  border: 1px solid rgba(0, 0, 0, 0.12);
  color: rgba(0, 0, 0, 0.54);
  width: 100%;
}

.emotion-1::-moz-focus-inner {
  border-style: none;
}

.emotion-1.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-1 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-1.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
  border: 1px solid rgba(0, 0, 0, 0.12);
}

.emotion-1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-1:hover {
    background-color: transparent;
  }
}

.emotion-1.Mui-selected {
  color: rgba(0, 0, 0, 0.87);
  background-color: rgba(0, 0, 0, 0.08);
}

.emotion-1.Mui-selected:hover {
  background-color: rgba(0, 0, 0, 0.12);
}

@media (hover: none) {
  .emotion-1.Mui-selected:hover {
    background-color: rgba(0, 0, 0, 0.08);
  }
}

.emotion-2 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<div
    class="MuiToggleButtonGroup-root MuiToggleButtonGroup-vertical MuiToggleButtonGroup-fullWidth emotion-0"
    data-testid="preferred-time-windows-tomorrow-options-group"
    role="group"
  >
    <button
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-fullWidth MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedVertical emotion-1"
      tabindex="0"
      type="button"
      value="[object Object]"
    >
      Morning (7am-12pm)
      <span
        class="MuiTouchRipple-root emotion-2"
      />
    </button>
    <button
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-fullWidth MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedVertical emotion-1"
      tabindex="0"
      type="button"
      value="[object Object]"
    >
      Afternoon (12-5pm)
      <span
        class="MuiTouchRipple-root emotion-2"
      />
    </button>
    <button
      aria-pressed="false"
      class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-fullWidth MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedVertical emotion-1"
      tabindex="0"
      type="button"
      value="[object Object]"
    >
      Evening (5-10pm)
      <span
        class="MuiTouchRipple-root emotion-2"
      />
    </button>
  </div>
</DocumentFragment>
`;
